File: 1404.number-of-steps-to-reduce-a-number-in-binary-representation-to-one/solution.go
func numSteps(s string) int {
    steps := 0
    carry := 0

    for i := len(s) - 1; i > 0; i-- {
        digit := int(s[i] - '0') + carry
        if digit == 1 {
            steps++
            carry = 1
        }
        steps++
    }

    return steps + carry
}

File: 1404.number-of-steps-to-reduce-a-number-in-binary-representation-to-one/solution.kt
class Solution {
    fun numSteps(s: String): Int {
        var steps = 0
        var carry = 0

        for (i in s.lastIndex downTo 1) {
            val digit = s[i] - '0' + carry
            if (digit == 1) {
                steps++
                carry = 1
            }
            steps++
        }

        return steps + carry
    }
}